### PR TITLE
Add regular expression matcher

### PR DESCRIPTION
**Pull request by ntl**

### CODE REVIEW

File: lib/identifier/uuid/uuid.rb [hunk 1]
```diff
@@ -1,5 +1,17 @@
 module Identifier
   module UUID
+    MATCHER = %r{
+      [a-f0-9]{8}          # abcdef01
+      -                    #         -
+      [a-f0-9]{4}          #          abcd
+      -                    #              -
+      4[a-f0-9]{3}         #               4abc
+      -                    #                   -
+      [89aAbB][a-f0-9]{3}  #                    8abc
```
Comment (sbellware) [inline lib/identifier/uuid/uuid.rb]: Wont the `i` make expressing both cases of the letters irrelevant?

File: lib/identifier/uuid/uuid.rb [hunk 2]
```diff
@@ -1,5 +1,17 @@
 module Identifier
   module UUID
+    MATCHER = %r{
```
Comment (sbellware) [inline lib/identifier/uuid/uuid.rb]: This appears to be a pattern, more than a `matcher`. Also, it's only a constant if it NEVER, EVER changes (like pi, or Planck's Constant).

File: lib/identifier/uuid/uuid.rb (hunk 1 above)
Comment (ntl) [inline lib/identifier/uuid/uuid.rb]: Good catch. Do we want insensitivity or sensitivity?

Comment (sbellware) [inline lib/identifier/uuid/uuid.rb]: UUIDs are allowed to be uppercase or lowercase, but only one or the other (I think). So, we could probably get away with insensitivity (I know I rely on it all the time :grin: )

File: lib/identifier/uuid/uuid.rb (hunk 2 above)
Comment (ntl) [inline lib/identifier/uuid/uuid.rb]: It should be a constant.

File: test/spec/matcher.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative './spec_init'
```
Comment (sbellware) [inline test/spec/matcher.rb]: Is this test testing code in the library?

File: lib/identifier/uuid/uuid.rb (hunk 1 above)
Comment (ntl) [inline lib/identifier/uuid/uuid.rb]: This is a "should" vs "must" decision then. I'm okay with conditional compliance here, as the regex for all-same case would be awkward (and likely require additional ruby code).

File: test/spec/matcher.rb (hunk 3 above)
Comment (ntl) [inline test/spec/matcher.rb]: No. The library does not use it.

**Comment by sbellware** [comment]
:red_circle: We have a problem on this repo. There's an unmerged branch from a couple days ago that alredy has the regex code.

**Comment by ntl** [comment]
Okay.

**Pull request closed without merging**